lib/Dialect/Handshake/Transforms/LowerExtmemToHW.cpp

```diff
@@ -156,8 +156,10 @@ LogicalResult HandshakeLowerExtmemToHWPass::wrapESI(
   llvm::SmallVector<unsigned> argReplacementsIdxs;
   llvm::transform(argReplacements, std::back_inserter(argReplacementsIdxs),
                   [](auto &pair) { return pair.first; });
-  for (auto i : llvm::reverse(argReplacementsIdxs))
-    wrapperModPortInfo.inputs.erase(wrapperModPortInfo.inputs.begin() + i);
+  for (auto i : llvm::reverse(argReplacementsIdxs)) {
+    wrapperModPortInfo.ports.erase(wrapperModPortInfo.begin_input() + i);
```

Comment:
it's generally bad practice to rely on code outside of a class to make necessary modifications to synchronized internal datastructures.

Reply:
Indeed.  I was trying to avoid having mutation options at all, but a few places were doing things like this.